crates/types/src/joints/mirror.rs

```diff
@@ -0,0 +1,3 @@
+pub trait Mirror {
+    fn mirrored(self) -> Self;
+}
```

Comment:
What do you think of moving all `impl Mirror for x<y>` to this file?

Reply:
I'd like to keep it at the individual types, as this is the usual way to implement traits for types we own.

Reply:
Okay